src/index.js

```diff
@@ -1,135 +1,161 @@
-							this.broker,
-							{
-								getJob: this.getJob,
-							}
-						), // context
-						false, // runOnInit
-						job.utcOffset || null, // utcOffset
-						job.unrefTimeout || null, // unrefTimeout
-					)
-					// Will be triggered at the start if it's defined
-					cronjob.runOnStarted = job.runOnInit;
-					cronjob.manualStart = job.manualStart || false
-					cronjob.name = job.name || this.makeid(20);
-					return cronjob;
-				};
-				
-				return instance_job;
-			});
-		}
-		return this.Promise.resolve();
-	},
-
-	events: {
-		"$broker.started": function() {
-			this.$crons = this.$crons.map((job, idx) => {
-				var jobCron = job();
-				this.$crons[idx] = jobCron; // sneaky
-				if (!jobCron.manualStart) {
-					jobCron.start();
-				}
-				this.logger.info(`Start Cron - '${jobCron.name}'`);
-
-				if (jobCron.runOnStarted) {
-					jobCron.runOnStarted();
-				}
-				return jobCron;
-			});
-		}
-	},
-
-	/**
-	 * Service stopped lifecycle event handler
-	 */
-	stopped() {
-		this.$crons.map((job) => {
-			job.stop();
-		});
-	}
-};
+  name: "cron",
+
+  settings: {
+    cronJobs: []
+  },
+
+  created() {
+    this.jobs = new Map();
+    this.validateAndCreateJobs();
+  },
+
+  started() {
+    this.startJobs();
+  },
+
+  async stopped() {
+    for (const job of this.jobs.values()) {
+      job.stopJob();
+    }
+  },
+
+  methods: {
+    validateAndCreateJobs() {
+      if (!Array.isArray(this.settings.cronJobs)) {
+        this.logger.warn("No cron jobs defined or invalid configuration");
+        return;
+      }
+
+      for (const jobConfig of this.settings.cronJobs) {
+        try {
+          this.createJob(jobConfig);
+        } catch (error) {
+          this.logger.error(`Error creating cron job: ${error.message}`, jobConfig);
+          console.error(error);
+        }
+      }
+    },
+
+    createJob(jobConfig) {
+      if (!jobConfig.name || !jobConfig.cronTime || !jobConfig.onTick) {
+        throw new Error("Invalid job configuration. Required: name, cronTime, onTick");
+      }
+    
+      try {
+        const job = new CronJob(
+          jobConfig.cronTime,
+          this.wrapOnTick(jobConfig.name, jobConfig.onTick),
+          this.wrapOnComplete(jobConfig.name, jobConfig.onComplete),
+          false,
+          jobConfig.timeZone,
+          this
+        );
+    
+        const jobWrapper = {
+          name: jobConfig.name,
+          cronJob: job,
+          startJob: function() {
+            if (jobConfig.OnStart) jobConfig.OnStart();
+            this.cronJob.start();
+          },
+          stopJob: function() {
+            if (jobConfig.OnStop) jobConfig.OnStop();
+            this.cronJob.stop();
+          },
+          lastDate: function() {
+            return this.cronJob.lastDate();
+          },
+          running: function() {
+            return this.cronJob.running;
+          },
+          setTime: function(time) {
+            return this.cronJob.setTime(new CronTime(time));
```

Comment:
I would lean towards using CronTime as the parameter to this (like in above suggestion extending the CronJobConfig to add `setTime`)
```suggestion
            return this.cronJob.setTime(time);
```

Reply:
Applied